src/sage/geometry/polyhedron/combinatorial_polyhedron/base.pyx

```diff
@@ -3631,14 +3585,45 @@ cdef class CombinatorialPolyhedron(SageObject):
         elif not do_edges and self._ridges is None:
             raise ValueError('could not determine ridges')
 
+    cdef bint _is_dual_faster_to_compute_edges_or_ridges(self, bint do_edges) except -1:
```

Comment:
I would suggest to use a more general method name that returns an "algorithm name" instead of a boolean

Reply:
Good point.

Taken care of in e4ea210f80c5f3c5bd95008a3f3174195b7184a8.

I think I optimized in all the right places. However, I also optimized in all the wrong places. Having testable methods with human-readable input and output seems better.